scripts/zkai_d128_activation_swiglu_proof_input.py

```
@@ -0,0 +1,628 @@
+
+def require_signed_m31(value: Any, label: str) -> None:
+    if not isinstance(value, int) or isinstance(value, bool):
+        raise ActivationSwiGluInputError(f"{label} must be an integer")
+    if value <= -M31_MODULUS or value >= M31_MODULUS:
+        raise ActivationSwiGluInputError(f"{label} outside signed M31 bounds")
+
+
+def load_source(path: pathlib.Path = SOURCE_JSON) -> dict[str, Any]:
+    try:
+        if not path.is_file():
+            raise ActivationSwiGluInputError(f"source gate/value evidence is not a regular file: {path}")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)